gensec/customPrecon.py

```diff
@@ -194,146 +429,156 @@ def vector_separation(cell_h, cell_ih, qi, qj):
     sij = np.dot(cell_ih, (qi - qj).T)  # column vectors needed
     sij -= np.rint(sij)
 
-    dij = np.dot(cell_h, sij).T         # back to i-pi shape
+    dij = np.dot(cell_h, sij).T  # back to i-pi shape
     rij = np.linalg.norm(dij, axis=1)
 
     return dij, rij
-    
-    
+
 
 def vdwHessian(atoms):
 
-#atoms = read(options.inputfile, format = options.formatfile)
+    # atoms = read(options.inputfile, format = options.formatfile)
 
-    N  = len(atoms)
+    N = len(atoms)
     coordinates = atoms.get_positions()
     atom_names = atoms.get_chemical_symbols()
     cell_h = atoms.get_cell()[:]
     cell_ih = atoms.get_reciprocal_cell()[:]
 
     def calculate_vdW(i, j, rij, coordinates, atom_names):
```

Comment:
not sure I understand the identention here and these nested definitions. why is this like this?

Reply:
make sure you take a careful look here